### PR TITLE
Add Win2D Interop sample

### PR DESCRIPTION
**Pull request by eigenvector32**
Add a sample that shows using noise being procedurally generated using Win2D along with other Windows.UI.Composition code. The addition to SamplesCommon/ImageLoader makes it so the surface created can be managed with the existing ManagedSurface codebase.

### CODE REVIEW

File: SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        }
+
+        private void RebuildButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+
+            var fireAndForget = SetupEffect();
```
Comment (robald) [inline SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs]: To enable dynamic updates from directly from the slider/combo updates, consider making SetupEffect() a one time initialization function for page load and create a separate UpdateSurface() method which is called in response to UI updates.

File: SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            _noiseEffect.Frequency = _noiseFrequency;
+            _noiseEffect.Noise = _noiseType;
+            _noiseEffect.Tileable = true;
+            _noiseEffect.Size = EffectSize;
+
+            _noiseLuminanceEffect = new LuminanceToAlphaEffect();
+            _noiseLuminanceEffect.Source = _noiseEffect;
+
+            _noiseSurface = ImageLoader.Instance.LoadInteropSurface(EffectSize.ToSize(), NoiseDrawHandler);
+            if (_noiseEffect == null)
```
Comment (robald) [inline SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs]: Did you intend to check _noiseSurface == null here?

File: SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+            {
+                DisposeEffect();
+                return;
+            }
+            _noiseSurface.Brush.Stretch = CompositionStretch.Fill;
+
+            var placeholderVisual = ElementCompositionPreview.GetElementVisual(Placeholder);
+            var compositor = placeholderVisual.Compositor;
+
+            _contentSurface = await ImageLoader.Instance.LoadFromUriAsync(new Uri("ms-appx:///Assets/Landscapes/Landscape-1.jpg"));
```
Comment (robald) [inline SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs]: You should avoid re-creating the contentSurface and Effects here every time the slider updates.  Please separate this into one time initialization of stuff that isn't affected by UI updates and only update the noise surface when the variables are changed.

File: SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+        }
+
+        private void ResizeEffect()
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+
+            _outputVisual.Size = EffectSize;
```
Comment (robald) [inline SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs]: You also need to resize the _noiseEffect in response to size changes.
_noiseEffect.Size = _outputVisual.Size;

File: SampleGallery/Shared/SampleDefinition.cs [hunk 5]
```diff
@@ -160,6 +160,7 @@ static public void RefreshSampleList()
...
                 new SampleDefinition(ShyHeader.StaticSampleName,                    typeof(ShyHeader),                    SampleType.EndToEnd,  SampleCategory.DynamicHumanInteractions,   false, false, dateAdded: new DateTime(2017,04,25),  imageUrl: "ms-appx:///Assets/SampleThumbnails/ShyHeader.PNG",               sdkVersion: RuntimeSupportedSDKs.SDKVERSION._14393, featured: true),
                 new SampleDefinition(BlurPlayground.StaticSampleName,               typeof(BlurPlayground),               SampleType.Reference, SampleCategory.Conceptual,                 true,  true,  "ms-appx:///Assets/SampleThumbnails/BlurPlayground.PNG",                                                          sdkVersion: RuntimeSupportedSDKs.SDKVERSION._14393),
+                new SampleDefinition(Win2DInterop.StaticSampleName,                 typeof(Win2DInterop),                 SampleType.Reference, SampleCategory.Conceptual,                 true,  true,  "ms-appx:///Assets/SampleThumbnails/Win2DInterop.PNG",                  sdkVersion: RuntimeSupportedSDKs.SDKVERSION._14393 ),
```
Comment (robald) [inline SampleGallery/Shared/SampleDefinition.cs]: Please sync to the latest samplegallery and merge with the latest changes.  There is a change required here for SampleType and a missing parameter.

File: SamplesCommon/SamplesCommon/ImageLoader/ImageLoader.cs [hunk 6]
```diff
@@ -163,6 +163,19 @@ public IAsyncOperation<ManagedSurface> LoadFromUriAsync(Uri uri, Size size, Load
...
 
+        public ManagedSurface LoadInteropSurface(Size size, InteropDrawHandler handler)
+        {
+            ManagedSurface surface = new ManagedSurface(CreateSurface(size));
+            var ignored = surface.Draw(_graphicsDevice, _drawingLock, new InteropDrawer(handler));
+
+            return surface;
+        }
+
+        public async Task RedrawSurface(ManagedSurface surface)
```
Comment (robald) [inline SamplesCommon/SamplesCommon/ImageLoader/ImageLoader.cs]: I don't see any callers of this.  Please remove.

File: SamplesCommon/SamplesCommon/ImageLoader/ManagedSurface.cs [hunk 7]
```diff
@@ -59,6 +59,11 @@ public CompositionDrawingSurface Surface
             get { return _surface; }
         }
 
+        public IContentDrawer Drawer
```
Comment (robald) [inline SamplesCommon/SamplesCommon/ImageLoader/ManagedSurface.cs]: I don't see any callers of this.  Please remove.

**Comment by daneuber** [comment]
Closing due to inactivity. Please open a new PR if you are still interested in merging this in.

**Pull request closed without merging**